### PR TITLE
PHOENIX-7379 Improve handling of concurrent index mutations with the …

### PR DESCRIPTION
**Pull request by kadirozde**
…same timestamp

### CODE REVIEW

**Comment by kadirozde** [comment]
This PR drastically reduces the number of times a thread sleeps (more than 6x) and visibly reduces total upsert time in ConcurrentMutationsExtendedIT#testConcurrentUpserts().

File: phoenix-core-server/src/main/java/org/apache/phoenix/hbase/index/IndexRegionObserver.java [hunk 1]
```diff
@@ -366,7 +365,8 @@ public int getMaxPendingRowCount() {
   private boolean shouldWALAppend = DEFAULT_PHOENIX_APPEND_METADATA_TO_WAL;
   private boolean isNamespaceEnabled = false;
   private boolean useBloomFilter = false;
-
+  private long lastTimestamp = 0;
```
Comment (stoty) [inline phoenix-core-server/src/main/java/org/apache/phoenix/hbase/index/IndexRegionObserver.java:368]: Longs are not atomic in Java.
Shouldn't this be atomic / volatile or be accessed in synchronized blocks everywhere ?

File: phoenix-core/src/it/java/org/apache/phoenix/end2end/ConcurrentMutationsExtendedIT.java [hunk 2]
```diff
@@ -357,6 +358,8 @@ public void testConcurrentUpserts() throws Exception {
         }
 
         assertTrue("Ran out of time", doneSignal.await(120, TimeUnit.SECONDS));
+        System.out.println("Total upsert time in ms : "
```
Comment (stoty) [inline phoenix-core/src/it/java/org/apache/phoenix/end2end/ConcurrentMutationsExtendedIT.java:361]: use LOGGER.info() instead ?

File: phoenix-core-server/src/main/java/org/apache/phoenix/hbase/index/IndexRegionObserver.java (hunk 1 above)
Comment (kadirozde) [reply phoenix-core-server/src/main/java/org/apache/phoenix/hbase/index/IndexRegionObserver.java:368]: It is accessed only in synchronized block and should be accessed only in synchronized block. So there is no need to define it as volatile or atomic, which will give the wrong impression.

Comment (stoty) [reply phoenix-core-server/src/main/java/org/apache/phoenix/hbase/index/IndexRegionObserver.java:368]: Thanks, you are right.

**Review by stoty — APPROVED** [review]
+1 LGTM

**Pull request merged**